cleanlab/segmentation/rank.py

```diff
@@ -0,0 +1,240 @@
+      
+    **kwargs:
+      downsample : int, optional
+      Factor to shrink labels and pred_probs by for 'num_pixel_issues' only . Default ``16``
+      Must be a factor divisible by both the labels and the pred_probs. Note that larger factors result in a linear 
+      decrease in performance
+
+
+
+    Returns
+    -------
+
+    image_scores: np.ndarray
+        Array of shape ``(N, )`` of scores between 0 and 1, one per image in the dataset.
+        Lower scores indicate image more likely to contain a label issue.
+    pixel_scores:
+        Array of shape ``(N,H,W)`` of scores between 0 and 1, one per pixel in the dataset.
+
+    """
+        
+    def check_input(labels: np.ndarray, pred_probs: np.ndarray) -> None:
+        if len(labels.shape) != 3:
+            raise ValueError("labels must have a shape of (N, H, W)")
+
+        if len(pred_probs.shape) != 4:
+            raise ValueError("pred_probs must have a shape of (N, K, H, W)")
+
+        num_images, height, width = labels.shape
+        num_images_pred, num_classes, height_pred, width_pred = pred_probs.shape
+
+        if num_images != num_images_pred or height != height_pred or width != width_pred:
+            raise ValueError("labels and pred_probs must have matching dimensions for N, H, and W")
+            
+    check_input(labels,pred_probs)
+    
+    softmin_temperature = kwargs.get("temperature", 0.1) 
+    downsample_num_pixel_issues = kwargs.get("downsample", 16) 
+
+    if method == "num_pixel_issues":
+        _,K,_,_ = pred_probs.shape
+        labels_expanded = labels[:, np.newaxis, :, :]
+        mask = (np.arange(K)[np.newaxis, :, np.newaxis, np.newaxis] == labels_expanded)
+        # Calculate pixel_scores
+        masked_pred_probs = np.where(mask, pred_probs, 0)
+        pixel_scores = masked_pred_probs.sum(axis=1)
+        
+        return find_label_issues(labels,pred_probs, downsample=downsample_num_pixel_issues, n_jobs=n_jobs, scores_only=True, verbose=verbose,), pixel_scores
+        
+    if downsample_num_pixel_issues != 16:
+        import warnings
+        warnings.warn(f"image will not downsample for method {method} is only for method: num_pixel_issues")
+        
+    num_im, num_class, h,w = pred_probs.shape
+    image_scores = []
+    pixel_scores = []
+    if verbose:
+        from tqdm.auto import tqdm
+        pbar = tqdm(desc=f"images processed using {method}", total=num_im)
+    for image in range(num_im):
+        mask = [labels[image]==cls for cls in range(num_class)]
+        image_probs = pred_probs[image][mask]
+        pixel_scores.append(np.where(mask, pred_probs[image], 0).sum(axis=0))
+        image_scores.append(get_label_quality_per_image(np.array(image_probs),method = method, temperature=softmin_temperature))
+        if verbose:
+            pbar.update(1)
+    return np.array(image_scores), np.array(pixel_scores)
+
+
+def get_label_quality_per_image(image_scores,method=None,temperature=0.1):
```

Comment:
Is this a private function? I do not see it in your tutorial. Add into tutorial if public, if private name it with private convention. Would be nice to clean up param typing here

Reply:
Also this might be mistitled, I think this returns a score for every image but reading the docstring and code it seems to return a single score for the dataset.